### DISCUSSION TITLE
Advice on how to get a sparse output from GNN?

### DISCUSSION THREAD

**Discussion by GabbySuwichaya** [pyg-team/pytorch_geometric#2451] [Q&A] [answered]
I would like to ask for your advice on training GNN for a link prediction task.

The problem that I have is that I tried to train the attention network.
The output is an adjacent matrix indicating the matches data.

I have tried to use the dropout layer at training to make my adjacent matrix to be sparse.

However, it seems that when I deactivate the dropout layer, my network performs very poorly. Could you give any tips on how to handle this kind of problem?

**Comment by rusty1s** [comment, marked as answer]
Can you clarify? I guess you have a `GAT` encoder and utilize a `InnerProduct` decoder as in https://github.com/rusty1s/pytorch_geometric/blob/master/examples/autoencoder.py? This will return a dense adjacency matrix by design, which you can sparsify via thresholding, e.g.
```
adj = adj > 0.5
```

**Reply by GabbySuwichaya** [reply]
Thank you. Also. I am so sorry for my late reply. 
Yes, you are right. The sparsity can be achieved by thresholding out the small value. 

I have just learned that I might have set the dropout layer too high. I will close this issue now.
